datafusion/common/src/scalar.rs

```diff
@@ -142,13 +143,13 @@ pub enum ScalarValue {
     /// Fixed size list scalar.
     ///
     /// The array must be a FixedSizeListArray with length 1.
-    FixedSizeList(ArrayRef),
+    FixedSizeList(Arc<FixedSizeListArray>),
```

Comment:
I know I'm coming in a little late, but have we considered instead making the type `Scalar<Arc<FixedSizeListArray>>`? ([Scalar being the one in arrow-rs](https://docs.rs/arrow/latest/arrow/array/struct.Scalar.html)) This would enforce that the length is 1, which might be nice. I could also understand if that makes it too awkward to work with.

Reply:
Personally I like the enforcement of length 1 (I mentioned this problem initially in the issue I opened), but it does seem to be a bit cumbersome to deal with.

Reply:
I think it is a good idea.

Reply:
Let's file a follow on ticket for continuing improving the representation of list constants